test/resolver.js

```diff
@@ -284,6 +284,14 @@ test('incorrect main', function (t) {
     });
 });
 
+test('missing main', function (t) {
+    var resolverDir = path.join(__dirname, 'resolver');
+    resolve('./missing_main', { basedir: resolverDir }, function (err, res, pkg) {
+        if (!err) t.fail('No error thrown when main was found missing.');
+        t.end();
```

Comment:
Another way of doing this would be:
```
if (err && err.code === 'INCORRECT_PACKAGE_MAIN') t.pass('Done');
t.end();
```

Which one would you prefer, @ljharb?

Reply:
I like tests that check more things - how about:
```js
        t.error(err, 'an error was thrown with a missing main');
        t.equal(err && err.code, 'INCORRECT_PACKAGE_MAIN', 'error has correct error code');
        t.end();
```

Reply:
`t.error(err, 'an error was thrown with a missing main');`

This line is causing the test to fail, probably because `t.error(obj, message, extra)` fails the assertion if `obj` is an error, according to the [docs](https://node-tap.org/docs/api/asserts/#terrorobj-message-extra).

Reply:
oh oops, in that case `t.ok(err instanceof Error` or something?